### ISSUE TITLE
Muting and then unmuting the video feed doesn't fully work

### ISSUE THREAD

**Issue by dwilson6** [jitsi/jitsi-meet#257]
Start a meeting with 2 users and then have one mute its video and and then enable the video again.  Then the other user cannot see the first user's video in the large video view.  It may work the first time but if you click on another video and click back then it will fail.  Interestingly the thumbnail video comes through fine.  

This is likely related to the new functionality which actually turns the camera off when it is muted (on https connections).  After some investigation it appears that the other endpoint never receives the streamended (in VIdeoLayout.js) or streamremoved (in JingleSession.js) event.  This causes the video element to not be removed so when the video is unmuted a second video element is added and then when the video container click handler runs it picks the first video element which is the old one to display in the full screen video.

**Comment by Zalmoxisus** [comment]
The interesting fact I've caught here, if the second participant also turns off and on his camera, then he will see the video of the second participant.

**Comment by dwilson6** [comment]
I haven't seen that behavior.  I have found that a temporary fix is to make sure that video stream elements are added to the beginning of the participant container.  In VideoLayout.addRemoteStreamElement, I changed the container.appendTo(streamElement) to container.insertBefore(streamElement, container.firstChild);  I might submit a pull request with this fix.  I haven't had a chance to look into why the events aren't being propagated to the other client.
